### ISSUE TITLE
Never expire by default

### ISSUE THREAD

**Issue by hyiltiz** [tomsmeding/pastebin-haskell#13]
Why not expire in 2 weeks by default? It is currently never expiring by default, which I think is probably relevant to the catall match here: 
https://github.com/tomsmeding/pastebin-haskell/blob/98d54d202c17ef28eed9024a6c9a51d2be49e344/Main.hs#L144

**Comment by tomsmeding** [comment]
Having a default of never expiring [was requested by `__monty__`](https://ircbrowse.tomsmeding.com/browse/haskell?id=229720&timestamp=1609500829#t1609500829) shortly after the expiration feature was proposed. Paraphrased, their reasoning is wanting to prevent link rot, and I agree with that reasoning, despite the fact that this results in ever-increasing disk usage on my side. :)

Do you have a particular reason for wanting temporary pastes by default?

Though that line is not the one that will produce the default behaviour in most cases; the fact that the `<select>` in [index.mustache](https://github.com/tomsmeding/pastebin-haskell/blob/master/index.mustache#L296) doesn't have a default option means that the top one is selected by default -- which is never.

**Comment by hyiltiz** [comment]
> Do you have a particular reason for wanting temporary pastes by default?
Yes, the right to be forgotten [0,1]. While users now have the right, intead of making it an opt-in feature, may as well make it an opt-out feature to prevent link rot. Most people will use pastebin services (and especially our users in #haskell) to get an answer to a question, then maybe someone reads up the IRC history for the past few days. Very old posts are probably irrelevant or the posters probably doesn't necessarily would prefer to keep them forever. The only incentive to prevent link rot is to serve a hypothetical reader of IRC who somehow dug up an old post, which is probably rare compared to common use, and they could probably re-ask a question if necessary. In some ways, it is similar to the GPL vs BSD: if default to never, current paster contributes a question along with its answer, it becomes a knowledge for all forever (like GPL); if default to expire, current paster asks and gets what he wants, and needs to make an active choice if the paster prefers the knowledge to stick around forever (or as long as @tomsmeding is bored or @simonmichael takes over the server, fullfilling the prophecy)... Anyway, enough philosophy and politics...


> Though that line is not the one that will produce the default behaviour in most cases; the fact that the `<select>` in [index.mustache](https://github.com/tomsmeding/pastebin-haskell/blob/master/index.mustache#L296) doesn't have a default option means that the top one is selected by default -- which is never.
Oh I had no idea that the program logic has creept into the "view" (of the MVC).

[0] https://papers.ssrn.com/sol3/papers.cfm?abstract_id=2689685
[1] https://gdpr.eu/right-to-be-forgotten/

**Comment by tomsmeding** [comment]
> Oh I had no idea that the program logic has crept into the "view" (of the MVC).

Where did I vouch for the neatness of the codebase structure? :)

Okay, onto the issue at hand. Prepare for some verbosity.

> Yes, the right to be forgotten [0,1].

Both the article (your [0]) and the GDPR ask for a right to be forgotten -- and while I don't have a public button on the paste site to automatically delete your data, I'm not going to deny requests. :)

The GDPR does say something about expiration of data (though not under the right to be forgotten); if I remember correctly, it says somewhere that personally identifiable data should not be kept longer than necessary for the original purposes of the data. Since I actually delete pastes that expire, I think we're complying here -- regardless of whether pastes are PI data or not.

Note that I'm responding to your literal points here, not necessarily your full story; see below.

> may as well make it an opt-out feature to prevent link rot.

I think I fail to see how making pastes expire by default is a means to prevent link rot. Do you mean that people will use a service that by default produces expiring links differently than a service that doesn't, resulting in less link-rot-producing behaviour? I don't think I find that likely in the case of a paste service, and especially this one: people will just use this to ask questions in #haskell, and won't really change their behaviour either way.

> The only incentive to prevent link rot is to serve a hypothetical reader of IRC who somehow dug up an old post, which is probably rare compared to common use, and they could probably re-ask a question if necessary.

Agreed, though we do have [ircbrowse](https://ircbrowse.tomsmeding.com/). Though I guess one can have opinions about public logging of IRC too; if we are to go there, I really have no substantial opinion, and perhaps you should discuss in #haskell first.

I do believe people have said about ircbrowse (perhaps `ski`?) that they used to link to earlier explanations of themselves in the past when a similar question was asked again.

> current paster asks and **gets what he wants**

(emphasis mine) That's a bold claim. :)

> (or as long as @tomsmeding is bored or `@simonmichael` takes over the server, fullfilling the prophecy)

Perhaps "never" is an incorrect description of that option, indeed. But a correct one would be far too verbose, and I think the intention is clear enough to those that care.

I'm not sure about your comparison of GPL/BSD, but let's suppose it's accurate to some extent. If "never expire by default" is GPL and "expire by default" is BSD, then -- notwithstanding the fact that pastebin-haskell is MIT at the moment, even weaker than BSD -- there are plenty of people that would choose for GPL, some quite fervently. Hence I don't think this comparison is really an argument to let pastes expire by default.

----

Having responded to your literal points, let me now say something with regards to your argument in full. You say that the argument _for_ keeping pastes indefinitely by default is weak, since people don't (shouldn't?) use IRC logs that way. Furthermore, you say that we have something called the right to be forgotten, and we can _strengthen_ that right to make it a _right to be forgotten soon by default_.

The first half I think you should ask to #haskell, not to me; I joined only last year, and I'm [not the most active user](https://ircbrowse.tomsmeding.com/nicks/haskell/all). \[1\] Additionally, I don't have strong opinions on matters like these; I'll submit to the opinion of #haskell.

As for the second half, I'm not sure I subscribe to the strengthening of the original right. Some people might agree, some people might not; I personally think it's unnecessary to strengthen the right, but then who am I. Here, too, I yield to the opinion of #haskell.

Conclusion: As stated I don't really have a strong opinion, but I nevertheless wanted to give a proper response to your argument. The bottom line, however, is: ask #haskell. :)

\[1\] When looking at that list, note that it is based on logs that started after I joined.

**Comment by hyiltiz** [comment]
Two take home messages:
- @tomsmeding is #6 on that list of active users in #haskell (just wow!) [well, it may be unfair to lambdabot to not count it/her/him]
- #haskell's pastebin is not run by dictatorship, not even a benevolent one.
